deltacat/io/memcached_object_store.py

```diff
@@ -47,8 +62,12 @@ def put(self, obj: object, *args, **kwargs) -> Any:
         serialized = cloudpickle.dumps(obj)
         uid = uuid.uuid4()
         current_ip = self._get_current_ip()
-        ref = self._create_ref(uid, current_ip)
-        client = self._get_client_by_ip(current_ip)
+        if self.storage_node_ips:
+            create_ref_ip = self._get_storage_node_ip(current_ip)
+        else:
+            create_ref_ip = current_ip
```

Comment:
Should we move this to a single function so that we don't have to have this if statement in every method?